audio/out/buffer.c

```diff
@@ -181,21 +205,26 @@ static int read_buffer(struct ao *ao, void **data, int samples, bool *eof)
 int ao_read_data(struct ao *ao, void **data, int samples, int64_t out_time_ns)
 {
     struct buffer_state *p = ao->buffer_state;
-    assert(!ao->driver->write);
 
     mp_mutex_lock(&p->lock);
 
-    int pos = read_buffer(ao, data, samples, &(bool){0});
+    int pos = ao_read_data_unlocked(ao, data, samples, out_time_ns, true);
```

Comment:
wouldn't this be better named `ao_read_data_locked`, since the lock is take in the previous line?

Reply:
This does indeed better match other similar uses in mpv.
My personal intuition is different, but I'll send a PR.